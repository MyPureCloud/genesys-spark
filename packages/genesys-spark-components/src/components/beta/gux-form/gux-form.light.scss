gux-form-beta {
  form {
    max-inline-size: var(--gse-semantic-formControl-form-maxWidth);

    header {
      display: flex;
      flex-direction: column;
      gap: var(--gse-semantic-formControl-formHeader-gap);
      padding-block-end: var(
        --gse-semantic-formControl-fieldset-header-paddingBottom
      );
      text-align: start;

      gux-form-heading {
        margin: 0;
        font-family: var(--gse-semantic-body-lg-semiBold-fontFamily) !important;
        font-size: var(--gse-semantic-body-lg-semiBold-fontSize) !important;
        font-weight: var(--gse-semantic-body-lg-semiBold-fontWeight) !important;
        line-height: var(--gse-semantic-body-lg-semiBold-lineHeight) !important;
        color: var(--gse-semantic-foreground-container-highEmphasis) !important;
      }
    }

    gux-form-description {
      margin: 0 !important;
      font-family: var(--gse-semantic-body-sm-regular-fontFamily) !important;
      font-size: var(--gse-semantic-body-sm-regular-fontSize) !important;
      font-weight: var(--gse-semantic-body-sm-regular-fontWeight) !important;
      line-height: var(--gse-semantic-body-sm-regular-lineHeight) !important;
      color: var(--gse-semantic-foreground-container-midEmphasis) !important;
    }

    fieldset {
      all: unset;
      display: flex;
      flex-direction: column;
      gap: var(--gse-semantic-formControl-formBody-gap);

      legend {
        display: flex;
        flex-direction: column;
        gap: var(--gse-semantic-formControl-fieldset-header-gap);
        padding-block-end: var(
          --gse-semantic-formControl-fieldset-paddingBottom
        );
        padding-inline: 0;
        text-align: start;

        gux-form-fieldset-heading {
          margin: 0;
          font-family: var(
            --gse-semantic-heading-md-bold-fontFamily
          ) !important;
          font-size: var(--gse-semantic-heading-md-bold-fontSize) !important;
          font-weight: var(
            --gse-semantic-heading-md-bold-fontWeight
          ) !important;
          line-height: var(
            --gse-semantic-heading-md-bold-lineHeight
          ) !important;
          color: var(
            --gse-semantic-foreground-container-highEmphasis
          ) !important;
        }
      }
    }

    /* Applies a 48px padding-bottom to any <fieldset> that is not immediately followed by a <gux-form-footer>. */
    fieldset:not(:has(+ gux-form-footer)) {
      padding-block-end: var(--gse-semantic-formControl-fieldset-paddingBottom);
    }

    /* Apply a margin-top of 16px to all other children within the form, except fieldset, gux-form-footer, and the first element after header or fieldset */
    > *:not(fieldset):not(gux-form-footer):not(header + *):not(fieldset + *) {
      margin-block-start: var(--gse-semantic-formControl-formBody-gap);
    }
  }

  /* Apply padding to all elements except the gux-form-footer where it is not needed */
  form > *:not(gux-form-footer) {
    padding-inline: var(--gse-semantic-formControl-form-margin);
  }
}
