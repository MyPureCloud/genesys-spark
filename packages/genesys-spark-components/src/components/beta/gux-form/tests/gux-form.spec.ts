import { newSpecPage } from '@test/specTestUtils';

import { GuxForm } from '../gux-form';

const components = [GuxForm];
const language = 'en';

describe('gux-form', () => {
  it('should build', async () => {
    const html = `
    <gux-form-beta>
    <form>
      <header>
        <gux-form-heading><h1 slot="heading">Form Heading</h1></gux-form-heading>
        <gux-form-description
          ><p slot="description">
            This is a simple paragraph to accompany the form header.
          </p></gux-form-description
        >
      </header>
      <fieldset>
        <legend>
          <gux-form-fieldset-heading
            ><h2 slot="heading">Form Fieldset Header</h2>
          </gux-form-fieldset-heading>
          <gux-form-description
            ><p slot="description">
              This is a simple paragraph to accompany the fieldset header.
            </p></gux-form-description
          >
        </legend>
        <gux-form-field-text-like label-position="above">
          <input slot="input" type="text" name="lp-3" />
          <label slot="label">Fieldset Form field label</label>
        </gux-form-field-text-like>

        <gux-form-field-phone label-position="above">
          <gux-phone-input-beta></gux-phone-input-beta>
          <label slot="label">Fieldset Form field label</label>
        </gux-form-field-phone>

        <gux-form-field-textarea label-position="above">
          <textarea slot="input" name="textarea"></textarea>
          <label slot="label">Fieldset Form field label</label>
        </gux-form-field-textarea>
      </fieldset>

      <gux-form-field-text-like label-position="above">
        <input slot="input" type="text" name="lp-3" />
        <label slot="label">Fieldset Form field label</label>
      </gux-form-field-text-like>

      <gux-form-footer placement="page-desktop">
        <footer>
          <gux-button accent="primary">Primary</gux-button>
          <gux-button accent="secondary">Secondary</gux-button>
        </footer>
      </gux-form-footer>
    </form>
  </gux-form-beta>
    `;
    const page = await newSpecPage({ components, html, language });

    expect(page.rootInstance).toBeInstanceOf(GuxForm);
  });

  describe('#render', () => {
    [
      `
      <gux-form-beta>
      <form>
        <header>
          <gux-form-heading><h1 slot="heading">Form Heading</h1></gux-form-heading>
          <gux-form-description
            ><p slot="description">
              This is a simple paragraph to accompany the form header.
            </p></gux-form-description
          >
        </header>
        <fieldset>
          <legend>
            <gux-form-fieldset-heading
              ><h2 slot="heading">Form Fieldset Header</h2>
            </gux-form-fieldset-heading>
            <gux-form-description
              ><p slot="description">
                This is a simple paragraph to accompany the fieldset header.
              </p></gux-form-description
            >
          </legend>
          <gux-form-field-text-like label-position="above">
            <input slot="input" type="text" name="lp-3" />
            <label slot="label">Fieldset Form field label</label>
          </gux-form-field-text-like>
        </fieldset>

        <gux-form-field-text-like label-position="above">
          <input slot="input" type="text" name="lp-3" />
          <label slot="label">Fieldset Form field label</label>
        </gux-form-field-text-like>

        <gux-form-footer placement="page-desktop">
          <footer>
            <gux-button accent="primary">Primary</gux-button>
            <gux-button accent="secondary">Secondary</gux-button>
          </footer>
        </gux-form-footer>
      </form>
    </gux-form-beta>
      `,
      `
      <gux-form-beta>
      <form>
        <header>
          <gux-form-heading><h1 slot="heading">Form Heading</h1></gux-form-heading>
          <gux-form-description
            ><p slot="description">
              This is a simple paragraph to accompany the form header.
            </p></gux-form-description
          >
        </header>
        <fieldset>
          <legend>
            <gux-form-fieldset-heading
              ><h2 slot="heading">Form Fieldset Header</h2>
            </gux-form-fieldset-heading>
            <gux-form-description
              ><p slot="description">
                This is a simple paragraph to accompany the fieldset header.
              </p></gux-form-description
            >
          </legend>
          <gux-form-field-text-like label-position="above">
            <input slot="input" type="text" name="lp-3" />
            <label slot="label">Fieldset Form field label</label>
          </gux-form-field-text-like>
        </fieldset>

        <gux-form-field-text-like label-position="above">
          <input slot="input" type="text" name="lp-3" />
          <label slot="label">Fieldset Form field label</label>
        </gux-form-field-text-like>
      </form>
    </gux-form-beta>
    `
    ].forEach((html, index) => {
      it(`should render gux-form component as expected (${index + 1})`, async () => {
        const page = await newSpecPage({ components, html, language });

        expect(page.root).toMatchSnapshot();
      });
    });
  });
});
