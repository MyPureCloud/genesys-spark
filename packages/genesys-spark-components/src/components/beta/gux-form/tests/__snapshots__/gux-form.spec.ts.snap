// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`gux-form #render should render gux-form component as expected (1) 1`] = `
<gux-form-beta>
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <form>
    <header>
      <gux-form-heading>
        <h1 slot="heading">
          Form Heading
        </h1>
      </gux-form-heading>
      <gux-form-description>
        <p slot="description">
          This is a simple paragraph to accompany the form header.
        </p>
      </gux-form-description>
    </header>
    <fieldset>
      <legend>
        <gux-form-fieldset-heading>
          <h2 slot="heading">
            Form Fieldset Header
          </h2>
        </gux-form-fieldset-heading>
        <gux-form-description>
          <p slot="description">
            This is a simple paragraph to accompany the fieldset header.
          </p>
        </gux-form-description>
      </legend>
      <gux-form-field-text-like label-position="above">
        <input slot="input" type="text" name="lp-3">
        <label slot="label">
          Fieldset Form field label
        </label>
      </gux-form-field-text-like>
    </fieldset>
    <gux-form-field-text-like label-position="above">
      <input slot="input" type="text" name="lp-3">
      <label slot="label">
        Fieldset Form field label
      </label>
    </gux-form-field-text-like>
    <gux-form-footer placement="page-desktop">
      <footer>
        <gux-button accent="primary">
          Primary
        </gux-button>
        <gux-button accent="secondary">
          Secondary
        </gux-button>
      </footer>
    </gux-form-footer>
  </form>
</gux-form-beta>
`;

exports[`gux-form #render should render gux-form component as expected (2) 1`] = `
<gux-form-beta>
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <form>
    <header>
      <gux-form-heading>
        <h1 slot="heading">
          Form Heading
        </h1>
      </gux-form-heading>
      <gux-form-description>
        <p slot="description">
          This is a simple paragraph to accompany the form header.
        </p>
      </gux-form-description>
    </header>
    <fieldset>
      <legend>
        <gux-form-fieldset-heading>
          <h2 slot="heading">
            Form Fieldset Header
          </h2>
        </gux-form-fieldset-heading>
        <gux-form-description>
          <p slot="description">
            This is a simple paragraph to accompany the fieldset header.
          </p>
        </gux-form-description>
      </legend>
      <gux-form-field-text-like label-position="above">
        <input slot="input" type="text" name="lp-3">
        <label slot="label">
          Fieldset Form field label
        </label>
      </gux-form-field-text-like>
    </fieldset>
    <gux-form-field-text-like label-position="above">
      <input slot="input" type="text" name="lp-3">
      <label slot="label">
        Fieldset Form field label
      </label>
    </gux-form-field-text-like>
  </form>
</gux-form-beta>
`;
